TweetMate/src/main/java/com/ybsystem/tweetmate/models/entities/twitter/TwitterStatus.java

```diff
@@ -0,0 +1,298 @@
+
+import com.ybsystem.tweetmate.application.TweetMateApp;
+import com.ybsystem.tweetmate.models.entities.Entity;
+import com.ybsystem.tweetmate.databases.PrefAppearance;
+import com.ybsystem.tweetmate.databases.PrefTheme;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import twitter4j.HashtagEntity;
+import twitter4j.MediaEntity;
+import twitter4j.Status;
+import twitter4j.URLEntity;
+import twitter4j.User;
+import twitter4j.UserMentionEntity;
+import twitter4j.util.TimeSpanConverter;
+
+import static androidx.core.text.HtmlCompat.TO_HTML_PARAGRAPH_LINES_INDIVIDUAL;
+import static com.ybsystem.tweetmate.resources.ResColor.*;
+
+@Data
+@EqualsAndHashCode(callSuper = false)
+public class TwitterStatus extends Entity {
+
+    private TwitterUser user;
+    private TwitterStatus rtStatus;
+    private TwitterStatus qtStatus;
+    private TwitterURLEntity[] urlEntities;
+    private TwitterMediaEntity[] mediaEntities;
+    private TwitterHashtagEntity[] hashtagEntities;
+    private TwitterUserMentionEntity[] userMentionEntities;
+
+    private Date createdAt;
+    private long id;
+    private long quotedStatusId;
+    private long currentUserRetweetId;
+
+    private String lang;
+    private String source;
+    private String text;
+    private int displayTextRangeStart;
+    private int displayTextRangeEnd;
+
+    // In reply to
+    private long inReplyToStatusId;
+    private long inReplyToUserId;
+    private String inReplyToScreenName;
+
+    // Count
+    private int favoriteCount;
+    private int retweetCount;
+
+    // State
+    private boolean isFavorited;
+    private boolean isRetweeted;
+    private boolean isRetweet;
+    private boolean isRetweetedByMe;
+
+    // Custom
+    private String convertedText;
+    private String convertedRelativeTime;
+    private String convertedAbsoluteTime;
+    private String convertedVia;
+    private ArrayList<String> imageUrls;
+    private boolean isMyTweet;
+    private boolean isPublic;
+    private boolean isTalk;
+    private boolean isDetail;
+    private boolean isTwiMedia;
+    private boolean isThirdMedia;
+    private boolean isVideo;
+
+    // Media
+    private static final Pattern PATTERN_YOUTUBE =
+            Pattern.compile("^https?://(?:www\\.youtube\\.com/watch\\?.*v=|youtu\\.be/)([\\w-]+)");
+
+    // Not use
+    // private Place place;
+    // private Scopes scopes;
+    // private GeoLocation geoLocation;
+    // private TwitterURLEntity quotedStatusPermalink;
+    // private TwitterSymbolEntity[] symbolEntities;
+    // private long[] contributorsIDs;
+    // private boolean isTruncated;
+    // private boolean isPossiblySensitive;
+
+    public TwitterStatus(Status status) {
+        User user = status.getUser();
+        Status rtStatus = status.getRetweetedStatus();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "rtStatus" which hides the field declared at line 39. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AHiddenFieldCheck)